bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -282,6 +279,95 @@ static int callAndWait(String[] pstr, ToIntFunction<IUnknown> callable) {
 	return phr[0];
 }
 
+class WebViewProvider {
+
+	private CompletableFuture<ICoreWebView2> webViewFuture = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_2> webView_2Future = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_11> webView_11Future = new CompletableFuture<>();
+
+	private CompletableFuture<Void> lastWebViewTask = webViewFuture.thenCompose(__ -> {
+		return CompletableFuture.allOf(webView_2Future.handle((result, ex) -> null), webView_11Future.handle((result, ex) -> null));
+	});
+
+
+	private ICoreWebView2 getWebView(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webViewFuture.join();
+	}
+
+	private void initializeWebView(ICoreWebView2 webView) {
+		webViewFuture.complete(webView);
+	}
```

Comment:
The name of this method is misleading, as it does not initialize the web view but just informs the provider that it is initialized. I propose to actually make this method initialize the webview, i.e., to have the according code here instead of the caller:

The name of this method is misleading, as it does not initialize the web view but just informs the provider that it is initialized. I propose to actually make this method initialize the webview, i.e., to have the according code here instead of the caller:
```suggestion
	private void initializeWebView(ICoreWebView2Controller controller) {
		long[] ppv = new long[1];
		controller.get_CoreWebView2(ppv);
		webViewFuture.complete(new ICoreWebView2(ppv[0]));
	}
```

Reply:
Yeah makes sense